src/IpAddress.php

```diff
@@ -69,11 +69,12 @@ class IpAddress implements MiddlewareInterface
      * @param array $trustedProxies   List of IP addresses of trusted proxies
      * @param string $attributeName   Name of attribute added to ServerRequest object
      * @param array $headersToInspect List of headers to inspect
+     * @throws \InvalidArgumentException
      */
     public function __construct(
-        $checkProxyHeaders = false,
+        bool $checkProxyHeaders = false,
         array $trustedProxies = null,
-        $attributeName = null,
+        string $attributeName = '',
```

Comment:
Changing to not accept null makes it backwards incompatible and will require a new major version number. I'm not sure that that's worth it for this change.

Reply:
I'm about to finish a PR which removes null for the `$trustedProxies` parameter too. I will update this branch and you will have a look on it, if you don't mind.